pkg/ingester/ingester.go

```diff
@@ -891,24 +891,39 @@ type pushStats struct {
 	perMetricSeriesLimitCount int
 }
 
+var pushReqCtxKey int
```

Comment:
This variable doesn't hold the key -- in fact it's the **pointer** to this variable that's the context key. While that works fine, it is different from the pattern that we use elsewhere (custom type + const value), I'd suggest to stick to existing pattern for consistency.